### PR TITLE
add cram/index support to dragmap

### PR DESCRIPTION
**Pull request by matthdsm**
- add cram and index output support to dragmap
- update dependent subworkflows

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/dragmap/align/main.nf [hunk 1]
```diff
@@ -45,9 +55,27 @@ process DRAGMAP_ALIGN {
...
     stub:
+    def args = task.ext.args ?: ''
+    def args2 = task.ext.args2 ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
+    def reads_command = meta.single_end ? "-1 $reads" : "-1 ${reads[0]} -2 ${reads[1]}"
+    def samtools_command = sort_bam ? 'sort' : 'view'
+    def extension_pattern = /(--output-fmt|-O)+\s+(\S+)/
+    def extension_matcher =  (args2 =~ extension_pattern)
+    def extension = extension_matcher.getCount() > 0 ? extension_matcher[0][2].toLowerCase() : "bam"
+    def reference = fasta && extension=="cram"  ? "--reference ${fasta}" : ""
```
Comment (itrujnara) [inline modules/nf-core/dragmap/align/main.nf:66]: Align equal signs

File: modules/nf-core/dragmap/align/main.nf [hunk 2]
```diff
@@ -10,10 +10,15 @@ process DRAGMAP_ALIGN {
...
     val   sort_bam
 
     output:
-    tuple val(meta), path("*.bam"), emit: bam
+    tuple val(meta), path("*.sam"), emit: sam   , optional: true
+    tuple val(meta), path("*.bam"), emit: bam   , optional: true
+    tuple val(meta), path("*.cram"), emit: cram , optional: true
+    tuple val(meta), path("*.crai"), emit: crai , optional: true
+    tuple val(meta), path("*.csi"), emit: csi   , optional: true
     tuple val(meta), path('*.log'), emit: log
```
Comment (itrujnara) [inline modules/nf-core/dragmap/align/main.nf:22]: Re-align commas for the emit column

**Review by itrujnara — COMMENTED** [review]
Some Harshil alignment issues spotted, otherwise it looks very good

File: modules/nf-core/dragmap/align/main.nf (hunk 1 above)
Comment (matthdsm) [reply modules/nf-core/dragmap/align/main.nf:66]: done

File: modules/nf-core/dragmap/align/main.nf (hunk 2 above)
Comment (matthdsm) [reply modules/nf-core/dragmap/align/main.nf]: fixed

**Pull request merged**